SSA/Projects/InstCombine/LLVM/Semantics.lean

```diff
@@ -156,10 +164,24 @@ def mul {w : Nat} (x y : IntW w) (flags : NoWrapFlags := {nsw := false , nuw :=
   let y' ← y
   let nsw := flags.nsw
   let nuw := flags.nuw
-  let AddSignedWraps? : Prop := nsw ∧
-    ((x'.toInt * y'.toInt) < -(2^(w-1)) ∨ (x'.toInt * y'.toInt) ≥ 2^w)
-  let AddUnsignedWraps? : Prop := nuw ∧ ((x'.toNat * y'.toNat) ≥ 2^w)
-  if (AddSignedWraps? ∨ AddUnsignedWraps?) then
+  let w1 := w-1
+  let w2 := 2*w
+  -- For mulitplication, we do the "naive" approach of doubling the size, doing the mulitplication, and comparing to the range.
+  -- Signed Wrap
```

Comment:
Can you potentially just compare the msb flags of the results. Also, check your comments for spelling.

Reply:
Maybe @luisacicolini has some ideas how we can avoid expanding all the way. Does Hacker's delight have some suggestions?

Reply:
I thought about it and took a look at the book, I don't think there are any simplifications. I would like to think about it some more and maybe give it a try myself over the next days and see if something comes up if that's okay :)

Reply:
We cam always merge this for now and replace it later with sth else.